File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/dto/UpdatePwdAdminDto.java
package com.TinyPro.entity.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

@Data
public class UpdatePwdAdminDto {

    @NotEmpty(message = "{validation.email.notEmpty}")
    @JsonProperty("email")
    private String email;

    @NotEmpty(message = "{validation.newPassword.notEmpty}")
    @Length(min = 6, message = "{validation.newPassword.length}")
    @JsonProperty("newPassword")
    private String newPassword;
    @JsonProperty("confirmNewPassword")
    private String confirmNewPassword; // 可选字段
}

File: template/nestJs/src/user/dto/update-pwd-admin.dto.ts
import { PartialType } from '@nestjs/mapped-types';
import { CreateUserDto } from './create-user.dto';
import { IsNotEmpty } from 'class-validator';
import { I18nTranslations } from '../../.generate/i18n.generated';
import { i18nValidationMessage } from 'nestjs-i18n';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class UpdatePwdAdminDto extends PartialType(CreateUserDto) {
  @ApiProperty({
    description: '目标用户的电子邮箱'
  })
  email: string;

  @ApiProperty({
    description: '新密码'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  newPassword: string;
  @ApiPropertyOptional({
    description: '确认密码'
  })
  confirmNewPassword?: string;
}
